packages/runtime/src/templates/handlerUtils.ts

```diff
@@ -201,6 +201,8 @@ export const getPrefetchResponse = (event: HandlerEvent, mode: string): HandlerR
       headers: {
         'Content-Type': 'application/json',
         'x-middleware-skip': '1',
+        // https://github.com/vercel/next.js/pull/42936/files#r1027563953
+        vary: 'x-middleware-prefetch',
```

Comment:
Good catch, what prompted the need for this?

Reply:
I'd not hit the problem, but saw that user's comment when investigating the feature. I could see it being an issue at some point